model/project_ref.go

```diff
@@ -247,16 +247,55 @@ func (p *ProjectRef) Add(creator *user.DBUser) error {
 	return p.AddPermissions(creator)
 }
 
+func (p *ProjectRef) GetRepoId() string {
+	return fmt.Sprintf("%s_%s", p.Owner, p.Repo)
+}
+
+func (p *ProjectRef) HasEditPermission(u *user.DBUser) bool {
+	if utility.StringSliceContains(p.Admins, u.Username()) {
+		return true
+	}
+	isBranchAdmin := u.HasPermission(gimlet.PermissionOpts{
+		Resource:      p.Id,
+		ResourceType:  evergreen.ProjectResourceType,
+		Permission:    evergreen.PermissionProjectSettings,
+		RequiredLevel: evergreen.ProjectSettingsEdit.Value,
+	})
+	if isBranchAdmin {
+		return true
+	}
+	// check if is repo admin
+	return u.HasPermission(gimlet.PermissionOpts{
+		Resource:      p.GetRepoId(),
+		ResourceType:  evergreen.ProjectResourceType,
+		Permission:    evergreen.PermissionProjectSettings,
+		RequiredLevel: evergreen.ProjectSettingsEdit.Value,
+	})
+}
+
 func (p *ProjectRef) AddPermissions(creator *user.DBUser) error {
 	rm := evergreen.GetEnvironment().RoleManager()
 	catcher := grip.NewBasicCatcher()
 	if !p.Restricted {
 		catcher.Wrapf(rm.AddResourceToScope(evergreen.UnrestrictedProjectsScope, p.Id), "error adding project '%s' to list of unrestricted projects", p.Id)
 	}
 	catcher.Wrapf(rm.AddResourceToScope(evergreen.AllProjectsScope, p.Id), "error adding project '%s' to list of all projects", p.Id)
+
+	flags, err := evergreen.GetServiceFlags()
```

Comment:
Shouldn't this always be run in case someone manages to create a branch even though the service flag is off?

Reply:
I  think resolved in person: no, because the repo ref might not exist yet,  the migration will have to handle adding resources